testsuite/MDAnalysisTests/core/test_topologyattrs.py

```diff
@@ -204,9 +220,18 @@ def test_set_residues(self):
 
 
 class TestResnames(TestResidueAttr):
-    values = np.array(['ARG', 'LYS', 'VAL', 'POPG'], dtype=np.object)
+    values = np.array(['VAL', 'LYS', 'VAL', 'POPG'], dtype=np.object)
     attrclass = tpattrs.Resnames
 
+    def test_get_named_residue(self):
```

Comment:
The `_get_named_residue` function should get called by a Segment(?) not the attr itself.  I think when the `Attr` is calling it the first argument (`self`) is the wrong class.  I think you might need to make a Universe to test this..